docs/developer-guide/plugins.md

```diff
@@ -66,6 +67,9 @@ ruleFunction.meta = meta;
 export default createPlugin(ruleName, ruleFunction);
 ```
 
+> [!NOTE]
+> The JSDoc annotation (`@type`) benefits auto-completion or type-check by TypeScript.
```

Comment:
minor

There is something wrong with this sentence.
If I had to guess you mean something like
```md
The JSDoc `@type` tag enables auto-completion and type-checking provided by Typescript.
```

Reply:
Looks good. Can you suggest code via **Add a suggestion** button? I'll take your suggestion. 😃 

<img width="207" alt="image" src="https://github.com/stylelint/stylelint/assets/473530/dbeaa576-a3b3-40f4-9949-b2b9f56c7fa5">

Reply:
I wanted to know if that what you meant first.

```suggestion
> The [JSDoc `@type` tag](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#type) enables the autocompletion and type-checking features provided by Typescript.
```